### PR TITLE
Test building troop ship and troop capacity

### PR DESCRIPTION
**Pull request by o01eg**
Test for https://github.com/freeorion/freeorion/issues/4409

Searches for Troop Ship design, enqueues it into production queue, and check troops capacity of built ship.

### CODE REVIEW

File: test/system/SmokeTestGame.cpp [hunk 1]
```diff
@@ -158,9 +165,45 @@ BOOST_AUTO_TEST_CASE(host_server) {
...
+
+            // enqueue Troop Ship
+            ScriptingContext context;
+
+            static constexpr boost::uuids::uuid troop_ship_uuid{0x08, 0xa5, 0x8b, 0x08, 0x09, 0x29, 0x49, 0x6d, 0x84, 0xfc, 0xfa, 0xa9, 0x14, 0x24, 0xca, 0x02};
+            for (const auto* design : GetPredefinedShipDesignManager().GetOrderedShipDesigns()) {
+                BOOST_TEST_MESSAGE("Predefined ship design " << design->Name() << " " << design->UUID());
+                if (design->UUID() == troop_ship_uuid) {
+                    BOOST_REQUIRE_EQUAL(troop_design_id, INVALID_DESIGN_ID);
+                    troop_design_id = my_empire->AddShipDesign(*design, GetUniverse());
```
Comment (geoffthemedio) [inline test/system/SmokeTestGame.cpp]: prefer `context` getters to free functions like `GetUniverse`

File: test/system/SmokeTestGame.cpp [hunk 2]
```diff
@@ -158,9 +165,45 @@ BOOST_AUTO_TEST_CASE(host_server) {
...
+                1,
+                home_planet->ID()),
+                context);
+        }
+
+        if (m_current_turn > 2) {
+            if (my_empire->GetProductionQueue().empty()) {
+                for (const auto* ship : Objects().findRaw<Ship>(is_owned)) {
+                    if (ship->DesignID() == troop_design_id) {
+                        BOOST_TEST_MESSAGE("Found troop ship with troops " << ship->TroopCapacity(GetUniverse()));
```
Comment (geoffthemedio) [inline test/system/SmokeTestGame.cpp]: same

Comment (o01eg) [reply test/system/SmokeTestGame.cpp]: Fixed.

File: test/system/SmokeTestGame.cpp (hunk 1 above)
Comment (o01eg) [reply test/system/SmokeTestGame.cpp]: Fixed.
Does `GetPredefinedShipDesignManager` free function have non-free version?

Comment (geoffthemedio) [reply test/system/SmokeTestGame.cpp]: No, since it has no gamestate in it.

**Pull request merged**